packages/frontend-app-api/src/tree/instantiateAppNodeTree.ts

```diff
@@ -289,12 +294,30 @@ export function createAppNodeInstance(options: {
         extensionDataRefs.add(ref);
       }
     } else if (internalExtension.version === 'v2') {
-      const outputDataValues = internalExtension.factory({
+      const context = {
         node,
         apis,
         config: parsedConfig,
         inputs: resolveV2Inputs(internalExtension.inputs, attachments),
-      });
+      };
+      const outputDataValues = options.extensionFactoryMiddleware
+        ? createExtensionDataContainer(
+            options.extensionFactoryMiddleware(
+              ({ config: configOverride } = {}) =>
+                createExtensionDataContainer(
+                  internalExtension.factory(
+                    configOverride
+                      ? {
+                          ...context,
+                          config: configOverride,
```

Comment:
The type indicates that that we can override inputs as well right? i.e. https://github.com/backstage/backstage/blob/4c46e99e2eac332e93243e0e656c97487548ac64/packages/frontend-plugin-api/src/wiring/createExtension.ts#L407-L412

I can take care of getting that sorted if you want though.